release-to-nuget has not published since 8.1.0.104: setup-dotnet cannot write to C:\Program Files\dotnet on the self-hosted runner

### Summary

Every release since `8.1.0.104` has been tagged and released on GitHub but **never published to
NuGet**. `8.1.0.105` and `8.1.0.106` both exist as tags and GitHub releases, while NuGet still serves
`8.1.0.104` as the newest version of the `clio` tool.

### Cause

The `release-to-nuget` job fails at its **second step**, `actions/setup-dotnet@v4`:

```
dotnet-install: The current user doesn't have write access to the installation root
'C:\Program Files\dotnet' to install .NET. Please try specifying a different installation
directory using the -InstallDir parameter, or ensure the selected directory has the
appropriate permissions.
##[error]Failed to install dotnet, exit code: 1.
```

It therefore never reaches `Extract Version from Tag`, `Test Solution`, `Pack clio with release
version`, `Verify packaged tool version` or `Publish to NuGet`.

That step was added to this workflow on 2026-08-14 by b5d6429c ("Speed up tests and run them on
NET10 only"), which copied the pattern from `build.yml`. `build.yml` runs on GitHub-hosted
`ubuntu-latest` / `windows-latest`, where the default install root is writable. `reliase-to-nuget.yml`
is the only workflow with `runs-on: self-hosted`, where the runner service account cannot write to
`C:\Program Files\dotnet`.

### Timeline

| Version | Publish run | Result |
|---|---|---|
| `8.1.0.104` | 2026-08-13 22:56 | success — **before** b5d6429c |
| b5d6429c adds `setup-dotnet` | 2026-08-14 01:14 | — |
| `8.1.0.105` | 2026-08-17, retried 2026-08-18 11:58 | failure at `setup-dotnet` |
| `8.1.0.106` | 2026-08-18 11:57 | failure at `setup-dotnet` |

### Impact

`dotnet tool install/update clio -g` cannot deliver anything newer than `8.1.0.104`, so merged fixes
do not reach users. This was found while trying to ship the fix for #1093 (PR #1094, merged as
`a4dd1f2b` and tagged `8.1.0.106`).

Note that re-running a failed publish replays the workflow file from the original run, so a workflow
fix requires a new tag to take effect.

### Suggested fix

Set `DOTNET_INSTALL_DIR` on the job to a writable location — the runner tool cache — which keeps the
NET10 requirement b5d6429c introduced (`dotnet test --framework net10.0`) without needing write access
to `Program Files`, and persists across runs so each release does not re-download both SDKs. PR to
follow.

Alternatives, if preferred: grant the runner account write access to `C:\Program Files\dotnet` (or
pre-install the 8.0 and 10.0 SDKs there), or drop the `setup-dotnet` step and rely on the runner's
pre-installed SDKs — the last only works if that runner already has .NET 10.

### Requested issue metadata

My permissions do not allow setting the issue type or labels. Requesting type **Bug** with labels
**bug** and **.NET** — could a maintainer apply them before review?

--- a/.github/workflows/reliase-to-nuget.yml
+++ b/.github/workflows/reliase-to-nuget.yml
@@ -16,6 +16,19 @@ jobs:
     # "running scripts is disabled on this system" (PSSecurityException / UnauthorizedAccess).
     env:
       PSExecutionPolicyPreference: Bypass
+      # actions/setup-dotnet installs into C:\Program Files\dotnet by default, which the
+      # self-hosted runner's service account cannot write:
+      #   "dotnet-install: The current user doesn't have write access to the installation root
+      #    'C:\Program Files\dotnet' to install .NET."
+      # The step was added to this workflow on 2026-08-14 (b5d6429c), copying the pattern from
+      # build.yml - but build.yml runs on GitHub-hosted ubuntu-latest/windows-latest, where the
+      # default root IS writable, whereas this job is the only one on self-hosted. Every release
+      # since has failed at that step before packing, so 8.1.0.105 and 8.1.0.106 were tagged and
+      # released on GitHub but never published to NuGet.
+      # Redirecting the install to the runner tool cache keeps the NET10 requirement that commit
+      # introduced (the test step below runs --framework net10.0) without needing write access to
+      # Program Files, and persists across runs so each release does not re-download the SDKs.
+      DOTNET_INSTALL_DIR: ${{ runner.tool_cache }}\dotnet
     steps:
       - name: Free disk space
         shell: powershell

